nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/DataTransferResource.java

```diff
@@ -133,15 +133,18 @@ protected void authorizeDataTransfer(final AuthorizableLookup lookup, final Reso
         }
 
         // get the authorizable
-        final DataTransferAuthorizable authorizable;
+        final Authorizable authorizable;
         if (ResourceType.InputPort.equals(resourceType)) {
-            authorizable = new DataTransferAuthorizable(lookup.getInputPort(identifier));
+            authorizable = lookup.getInputPort(identifier);
         } else {
-            authorizable = new DataTransferAuthorizable(lookup.getOutputPort(identifier));
+            authorizable = lookup.getOutputPort(identifier);
         }
 
         // perform the authorization
-        authorizable.authorize(authorizer, RequestAction.WRITE, user);
+        final PortAuthorizationResult authorizationResult = ((RootGroupPort) authorizable).checkUserAuthorization(user);
```

Comment:
@mcgilman Thanks for the update! Now the purpose of RootGroupPortAuthorizable is much clear :)
+1 for the proposal.